applications/OptimizationApplication/custom_utilities/response/max_overhang_response_utils.cpp

```diff
@@ -0,0 +1,274 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   license: OptimizationApplication/license.txt
+//
+//  Main author:     Reza Najian Asl
+//
+
+
+// System includes
+#include <tuple>
+#include <sstream>
+
+// Project includes
+#include "includes/model_part.h"
+#include "utilities/parallel_utilities.h"
+#include "utilities/reduction_utilities.h"
+#include "utilities/variable_utils.h"
+#include "utilities/atomic_utilities.h"
+#include "utilities/openmp_utils.h"
+
+// Application includes
+#include "custom_utilities/geometrical/model_part_utils.h"
+#include "optimization_application_variables.h"
+
+// Include base h
+#include "max_overhang_response_utils.h"
+
+namespace Kratos
+{
+
+double MaxOverhangAngleResponseUtils::CalculateValue(
+    const std::vector<ModelPart*>& rModelParts, 
+    const Parameters ResponseSettings)
```

Comment:
As we discussed, I would make this class non-static and pass the `ResponseSettings` only once. In this case, I would use `ValidateAndAssignDefaults` method to validate the settings. So the python version of this will be merely a empty hull calling methods of this class.

Reply:
as discussed I will pass Parameters as a copy.